src/lib/utils/getContainerNode.ts

```diff
@@ -23,7 +23,7 @@ function getContainerNode(selectorOrHTMLElement: string | HTMLElement) {
     throw new Error(errorMessage);
   }
 
-  return domElement;
+  return domElement as HTMLElement;
```

Comment:
You can avoid the cast with `querySelector<HTMLElement>` inside the function.

Reply:
Then the signature becomes:

```js
getContainerNode(selectorOrHTMLElement: string | HTMLElement): HTMLElement | null
```

While we don't want (and it can't) be `null`.

Reply:
With a proper typing of `isDOMElement` we should not have the issue, we can ensure that at this stage the selector is a DOM element.

Reply:
Yep, looks like `isDomElement` should be a type guard: https://www.typescriptlang.org/docs/handbook/advanced-types.html#type-guards-and-differentiating-types

(and the post by @wittydeveloper [medium](https://medium.com/@wittydeveloper/typescript-make-types-real-the-type-guard-functions-814364e8dbe3))

Reply:
Indeed, that's what we're looking for!